### PR TITLE
rm: Add doc on Update Rollback

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 51b07a476f19c380da628f1cdfa64053ec4fde4e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1366/docs/artifacts/html/index.html

File: source/reference-manual/ota/update-rollback.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If a new Target installation includes just Apps update, and new version Apps creation or start fails then the rollback process does the following:
+
+1. Aktualizr-lite (re-)creates and starts the previous version of Apps' containers.
+2. Aktualizr-lite prunes unused Compose Apps data (container images/layers of the failed new version).
+
+
+.. _Target:
+  https://foundries.io/insights/blog/2020/05/14/whats-a-target
```
Comment (doanac) [inline source/reference-manual/ota/update-rollback.rst:92]: if we can get this merged:

 https://github.com/foundriesio/docs/pull/345

then we don't have to link outside our docs

File: source/reference-manual/ota/update-rollback.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+.. _OSTree:
+  https://github.com/ostreedev/ostree
+.. _Compose Apps:
+  https://docs.foundries.io/latest/reference-manual/docker/compose-apps.html
+.. _OSTree Deployment Doc:
+  https://ostreedev.github.io/ostree/deployment/
+.. _aktualizr-lite parameter:
+  https://docs.foundries.io/latest/reference-manual/ota/aktualizr-lite.html#parameters
+.. _Bootloader update:
+  https://docs.foundries.io/latest/reference-manual/security/boot-software-updates-imx.html#update-procedure
```
Comment (doanac) [inline source/reference-manual/ota/update-rollback.rst:104]: we can use actual RST references to these docs.fio things. That will generate the proper links and work better with released versions of docs.

**Review by doanac — COMMENTED** [review]
I just read for technical details. It makes sense. There's some grammar we can improve, but I don't think that's your concern yet (and we can have someone else review that aspect)

File: source/reference-manual/ota/update-rollback.rst (hunk 2 above)
Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst:104]: ok, I just don't see such the approach in the other documents.

File: source/reference-manual/ota/update-rollback.rst (hunk 1 above)
Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst:92]: I've read the new doc, it's rather about *condensed* targets. The blog explains a notion of `Target` in general which is exactly what is needed in this context. Unfortunately, I haven't found a good definition of `Target` in our docs.

File: source/reference-manual/ota/update-rollback.rst (hunk 2 above)
Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst]: ok

**Comment by doanac** [comment]
Docs for 36a265dbc18b11ddc616fa6a42571b8f23effffe are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1374/docs/artifacts/html/index.html

File: source/reference-manual/ota/update-rollback.rst (hunk 1 above)
Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst:91]: I've just checked the PR #345 more carefully, yes, your're right, it makes sense to reference it from this doc. For some reason, I saw just part of the doc when I read it for the first time :).

Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst:91]: I'll update my PR once #345 is merged.

**Comment by mike-sul** [comment]
@igoropaniuk Please, check out whether p. `Rollback driven by a bootloader update failure` makes sense to you.

**Comment by doanac** [comment]
Docs for 50a1e05984e0cfde3e6cbf7b739fcf7df8a5d055 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1382/docs/artifacts/html/index.html

File: source/reference-manual/ota/update-rollback.rst (hunk 1 above)
Comment (mike-sul) [reply source/reference-manual/ota/update-rollback.rst]: done

**Review by kprosise — APPROVED** [review]
Suggestions provided.

**Comment by mike-sul** [comment]
@kprosise Updated the PR according to your comments.

**Comment by doanac** [comment]
Docs for 8b48c43d137d0bdfba291b3ccb5617dae5e15e97 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1383/docs/artifacts/html/index.html

**Review by MrCry0 — APPROVED** [review]

**Comment by kprosise** [comment]
> The `link-check` test has failed because `https://u-boot.readthedocs.io/en/latest/usage/scp03.html` refers to a non-existing page. I suppose it can be fixed in another PR?

Fixed earlier today.

**Pull request merged**